doc/anatomy_of_an_extension.md

```diff
@@ -0,0 +1,236 @@
+## Anatomy of an Extension Package
+Each extension package contains a single file with **metadata** about the extension and one or more **components**. 
+This is a simplified diagram of the folder structure of an extension package: 
+```
+extension-package/
+├── metadata.json
+└── components/
+    ├── component_A/
+    │   ├── doc/
+    │   ├── test/
+    │   ├── src/
+    │   │   └── procedure.sql
+    │   └── metadata.json
+    └── component_B/
+        ├── doc/
+        ├── test/
+        ├── src/
+        │   └── procedure.sql
+        └── metadata.json
+```
+___
+
+### Extension's metadata
+The extension's metadata file defines attributes like its name, the group or category the extension belongs, version, author, description, icon, etc. 
+
+Extension's metadata is defined in a [`metadata.json`](../metadata.json) file in the root folder of this repo.
+
+In that file, you will see that there is a `details` array  that accepts different custom objects defined by `"name"` and `"value"` properties. 
+
+These details will render in the CARTO UI when displaying the extension details. This is an example: 
+```json
+"details": [
+    {
+        "name": "License",
+        "value": "Apache 2.0"
+    },
+    {
+        "name": "Custom detail",
+        "value": "Value for custom detail"
+    }
+]
+```
+
+There is also a `components` object that should contain an array of all the components included in the extension. For example: 
+```json
+"components": [
+    "my_custom_component",
+    "another_custom_component"
+]
+```
+___
+
+### Components
+
+Components should aim to cover specific functionality, like adding a new column with a UUID, running an analysis and storing the result on a separate row or sending an HTTP request to a specific endpoint. 
```

Comment:
I don't think this example (storing in a separate row) is a good one. Components should generate new tables, not modify input ones

Reply:
oops, good catch. I meant column not row 🙈